homeassistant/components/dsmr/config_flow.py

```diff
@@ -87,11 +87,17 @@ def update_telegram(telegram: dict[str, DSMRObject]) -> None:
             if self._equipment_identifier in telegram:
                 self._telegram = telegram
                 transport.close()
+
             # Swedish meters have no equipment identifier
             if self._dsmr_version == "5S" and obis_ref.P1_MESSAGE_TIMESTAMP in telegram:
                 self._telegram = telegram
                 transport.close()
 
+            # gas meters
+            if obis_ref.EQUIPMENT_IDENTIFIER_GAS in telegram:
+                self._telegram = telegram
+                transport.close()
```

Comment:
It seems there's a risk we may call `transport.close` more than once, I think we should rewrite this function:
```py
        def update_telegram(telegram: dict[str, DSMRObject]) -> None:
            telegram_valid = False

            if self._equipment_identifier in telegram:
                telegram_valid = False
            # Swedish meters have no equipment identifier
            elif self._dsmr_version == "5S" and obis_ref.P1_MESSAGE_TIMESTAMP in telegram:
                telegram_valid = True
            # Allow meter with only gas equipment identifier
            elif obis_ref.EQUIPMENT_IDENTIFIER_GAS in telegram:
                telegram_valid = True

            if not telegram_valid:
                return

            self._telegram = telegram
            transport.close()
```